### ISSUE TITLE
json_error_max → JSON_ERROR_MAX

### ISSUE THREAD

**Issue by DimitriPapadopoulos** [json-parser/json-parser#124]
I would like to change `json_error_max` to uppercase, this is common C usage.

Is it considered to be part of the API? If not, it could be moved to `json.c`, otherwise we could keep both.

**Comment by DimitriPapadopoulos** [comment]
Thinking about it, it has to be part of the API, for the end user to be able to provision a large enough `error_buf`.

**Comment by LB--** [comment]
Yes, I added it in 2013 in 09896ab92be35a5e6039cb1d0715fcc0fb3d0fd1 for specifically that reason. I probably should have made it all caps, but back during 2013 there was no precedent for that in the codebase, and `json_value_none` was lowercase, so I matched that. It can certainly be changed to better clarify its constant nature.
